### PR TITLE
test: add unit test for empty input

### PR DESCRIPTION
**Pull request by einoyrjanainen**
## Type of change

- [ ] Bug fix
- [ ] New feature
- [ ] Other

- [ ] Non-breaking change
- [ ] Breaking change

## Developer checklist 

- [ ] A [CHANGELOG entry] has been included (only when change is visible to users)

[CHANGELOG entry]: https://github.com/nlsfi/geogen-algorithms/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by einoyrjanainen** [comment]
analyze.py lisätty sisältäen enismmäisenä funktionsa (calculate_coverage)
lisäksi test_analyze.py sisältäen kolme funktiota yksikkötestausta varten

File: src/geogenalg/analyze.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        .rename(columns={"intersect_area": "overlay_area"})
+    )
+
+    base_features = base_features.merge(
+        combined_overlay_area, on="base_feature_id", how="left"
+    )
+    base_features["overlay_area"] = base_features["overlay_area"].fillna(0)
+    base_features[coverage_attribute] = (
+        100 * base_features["overlay_area"] / base_features["base_area"]
+    )
```
Comment (rnousia) [inline src/geogenalg/analyze.py:56]: Extra attributes (base_feature_id, overlay_area, base_area) should probably be dropped after merge as they are not needed anymore and extra attributes increase the size of the dataframe. Also, the docstring only mentions that coverage attribute is added so that's what the caller of the function expects.

File: test/test_analyze.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                },
+                geometry=[
+                    Polygon([(0, 0), (1, 0), (1, 1), (0, 1), (0, 0)]),
+                ],
+            ),
+        )
+
+
+# kolmantena testi tyhjälle dataframelle
+def test_calculate_coverage_empty_input():
```
Comment (rnousia) [inline test/test_analyze.py:63]: Remove the comment. The test function name could be changed to `test_calculate_coverage_using_empty_input` as it's more readable?

File: test/test_analyze.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                "class": ["a"],
+                "id": [1],
+            },
+            geometry=[
+                Polygon([(0, 0), (3, 0), (3, 3), (0, 3), (0, 0)]),
+            ],
+            crs="EPSG:3067",
+        ),
+    )
+
```
Comment (rnousia) [inline test/test_analyze.py:37]: Test is missing asserts, there should be at least two:
- assert that "coverage" is in resulting geodataframe columns
- assert that overlay_features["coverage"] has the correct value

File: test/test_analyze.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+# kolmantena testi tyhjälle dataframelle
+def test_calculate_coverage_empty_input():
+    overlay = GeoDataFrame({"id": []}, geometry=[], crs="EPSG:3067")
+    base = GeoDataFrame({"class": [], "id": []}, geometry=[], crs="EPSG:3067")
+
+    result = calculate_coverage(overlay, base)
+
+    assert result.empty
```
Comment (rnousia) [inline test/test_analyze.py:69]: You could also add assert for checking that "coverage" is added to the result (if that's how the function is supposed to function?)

**Review by rnousia — CHANGES_REQUESTED** [review]
